src/lib/selections.ts

```diff
@@ -101,38 +109,74 @@ export async function getEventForSelectWithPoint(
       resp?.resp?.data?.modeling_response?.type === 'entity_get_parent_id'
         ? resp?.resp?.data?.modeling_response?.data?.entity_id
         : ''
-    const parentArtifact = engineCommandManager.artifactMap[parentId]
+    const parentArtifact = engineCommandManager.artifactGraph.get(parentId)
     if (parentArtifact) {
       _artifact = parentArtifact
     }
   }
-  const sourceRange = _artifact?.range
   if (_artifact) {
-    if (_artifact.type === 'extrudeCap')
+    if (_artifact.type === 'solid2D') {
+      const codeRef = getSolid2dCodeRef(
+        _artifact,
+        engineCommandManager.artifactGraph
+      )
+      if (err(codeRef)) return null
+      return {
+        type: 'Set selection',
+        data: {
+          selectionType: 'singleCodeCursor',
+          selection: { range: codeRef.range, type: 'solid2D' },
+        },
+      }
+    }
+    if (_artifact.type === 'cap') {
+      const codeRef = getCapCodeRef(
+        _artifact,
+        engineCommandManager.artifactGraph
+      )
+      if (err(codeRef)) return null
       return {
         type: 'Set selection',
         data: {
           selectionType: 'singleCodeCursor',
           selection: {
-            range: sourceRange,
-            type: _artifact?.cap === 'end' ? 'end-cap' : 'start-cap',
+            range: codeRef.range,
+            type: _artifact?.subType === 'end' ? 'end-cap' : 'start-cap',
           },
         },
       }
-    if (_artifact.type === 'extrudeWall')
+    }
+    if (_artifact.type === 'wall') {
+      const codeRef = getWallCodeRef(
+        _artifact,
+        engineCommandManager.artifactGraph
+      )
+      if (err(codeRef)) return null
       return {
         type: 'Set selection',
         data: {
           selectionType: 'singleCodeCursor',
-          selection: { range: sourceRange, type: 'extrude-wall' },
+          selection: { range: codeRef.range, type: 'extrude-wall' },
+        },
+      }
+    }
+    if (_artifact.type === 'segment') {
+      return {
+        type: 'Set selection',
+        data: {
+          selectionType: 'singleCodeCursor',
+          selection: { range: _artifact.codeRef.range, type: 'default' },
+        },
+      }
+    }
+    if (_artifact.type === 'path') {
+      return {
+        type: 'Set selection',
+        data: {
+          selectionType: 'singleCodeCursor',
+          selection: { range: _artifact.codeRef.range, type: 'default' },
```

Comment:
Nit: Could combine these into one if statement but I understand the practically of keeping them separate

Reply:
Thanks

c21a468a480bf4584d5044d0611388e47ea18b30

I started to realise I needed to get this merged and move on because I've been looking at the same code for too long at this point.